test/linear_eq.jl

```diff
@@ -0,0 +1,20 @@
+using IntervalArithmetic, StaticArrays, IntervalRootFinding
+
+@testset "Linear Equations" begin
+
+    A = [[2..3 0..1;1..2 2..3], ]
+    b = [[0..120, 60..240], ]
+    x = [[-120..90, -60..240], ]
+
+
+    for i in 1:length(A)
+        for precondition in (false, true)
+            for f in (gauss_seidel_interval, gauss_seidel_contractor, gauss_elimination_interval, \)
+                @test all(x[i] .⊆ f(A[i], b[i]))
+            end
+        end
+    end
+end
+
+\(A::StaticMatrix{Interval{T}}, b::StaticArray{Interval{T}}; kwargs...) where T = gauss_elimination_interval(A, b, kwargs...)
```

Comment:
These should be moved to the source code.

I think you can define a single method if you use `AbstractMatrix` and `AbstractVector`.

Reply:
>These should be moved to the source code.

I'm terribly sorry, I didn't realize.

>I think you can define a single method if you use AbstractMatrix and AbstractVector

I tried `AbstractArray`s earlier, but it doesn't work since it matches its type with the `\` for `StaticArray`s or `Array`s instead.

>What's the advantage of preconditioning, then? I thought it was supposed to narrow, not widen, the result

Preconditioning doesn't always give a better result. Specifically, for [M-matrices](https://en.wikipedia.org/wiki/M-matrix), the non-preconditioned version computes the exact hull whereas preconditioning causes some widening.